### ISSUE TITLE
Armbian does not bootup from EMMC on S96MAX+_2101

### ISSUE THREAD

**Issue by drafort** [ophub/amlogic-s9xxx-armbian#1144]
**Device Information | 设备信息**
 - SOC: S0905x3
 - Model [x96max_plus_2101 as listed in issue #1086 ]

**Armbian Version | 系统版本**
 - Kernel Version: [6.1.15]
 - Release: [jammy/bullseye]

**Describe the bug | 问题描述**

Thank you for support @ophub and @salsabilng!

Previously system was booted from usb (bluetooth is working).

After successful installing to EMMC with "armbian-install" command, removing usb and reinserting power supply I have black screen - no boot.
![Screenshot_20230307_235715](https://user-images.githubusercontent.com/95050279/223554288-16136cd9-4999-467e-9b3f-0a48a82b7bd7.jpg)
![Screenshot_20230307_235747](https://user-images.githubusercontent.com/95050279/223554357-5df091f1-ce05-4904-988d-10b121fc5b5e.jpg)

**Comment by Travis90x** [comment]
Same issue with kernel 6.1.15, 
so I tried 5.15.98_server_2023.03.06 and I can boot correctly with eMMC after installation.

My model is 2101W with W522A radio module (Fn-Link K255B-SR-B) 
and Wifi + BT still not recognized with 2101 DTB file (Fn-Link 6222B-SR C).

I tried to boot with meson-sm1-x96-air.dtb (because it is advised for model 100W with W522A)
and got the error UUID does not exist.
I connected sdcard with USB card reader to USB 3.0 port
blkid now can see also
/dev/sda1 : LABEL_FATBOOT="BOOT"…
/dev/sda2 : LABEL="ROOTFS" UUID="correct UUID" (corresponding to UUID in uEnv.txt)
and I still cannot boot, but I can boot if use 
APPEND=root=LABEL=ROOTFS in uEnv.txt

I tried meson-sm1-x96-air-gbit.dtb too, 
but wifi & BT still not recognized.
![S905X3 2101W_2](https://user-images.githubusercontent.com/23300077/223617688-da93c3c8-ec16-41f8-a229-373365b62711.jpg)

**Comment by Travis90x** [comment]
Try another DTS file like meson-sm1-x96-max-plus.dtb 
Now I discover that with meson-sm1-x96-max-plus-2101.dtb
I cannot power on the clock LED, but with meson-sm1-x96-max-plus.dtb
armbian-openvfd works well.

**Comment by drafort** [comment]
> Try another DTS file like meson-sm1-x96-max-plus.dtb Now I discover that with meson-sm1-x96-max-plus-2101.dtb I cannot power on the clock LED, but with meson-sm1-x96-max-plus.dtb armbian-openvfd works well.

Bluetooth is working for me (from usb) only with 2101.dtb, suppose, for installation on EMMC I also need 2101.dtb

**Comment by salsabilng** [comment]
2101W has different radio module, 2101 uses RTL8822cs,

**Comment by Travis90x** [comment]
> x96-max-plus.dtb

yes, of course I tested all the dts sm1-x96-max-plus*, for now only LAN 1000 works when not limited.

**Comment by salsabilng** [comment]
> > x96-max-plus.dtb
> 
> yes, of course I tested all the dts sm1-x96-max-plus*, for now only LAN 1000 works when not limited.

W522A module doesnt have any mainline driver for now, means wifi and bluetooth will not work, since your gbit eth work, just use eth for netwoking.

**Comment by drafort** [comment]
> 

Thank you @salsabilng, successfully installed on EMMC with x96-max-plus.dtb and mainline boot, but.... without Bluetooth support again (((
What to do?)

![Screenshot_20230309_221852](https://user-images.githubusercontent.com/95050279/224131800-d0f8cb99-cb1a-4ac4-b823-25be7cd9ce59.jpg)

**Comment by salsabilng** [comment]
`sudo nano /boot/uEnv.txt` edit dtb line to x96-max-plus-2101.dtb. `Ctrl + O` then enter, `Ctrl + X`. reboot

**Comment by drafort** [comment]
> 

@salsabilng, big thanks! it woks!!!!
![Screenshot_20230310_062118](https://user-images.githubusercontent.com/95050279/224215237-db6c79a8-05e3-4e45-ad56-5b290d34de6b.jpg)

**Comment by salsabilng** [comment]
looks like the armbian-install script has a problem when installing using 2101 @ophub , anyway great job! @drafort keep in mind, only 6.1.y works for bluetooth

**Comment by ophub** [comment]
https://github.com/ophub/amlogic-s9xxx-armbian/blob/570c7ae5086938f3d76beee9f5cc00ad6cd8d7c0/build-armbian/armbian-files/common-files/etc/model_database.conf#L145

select: `524`

**Comment by drafort** [comment]
> https://github.com/ophub/amlogic-s9xxx-armbian/blob/570c7ae5086938f3d76beee9f5cc00ad6cd8d7c0/build-armbian/armbian-files/common-files/etc/model_database.conf#L145
> 
> select: `524`
> 
> Didn't add `mainline_uboot`, have you tested that it works?
> 
> x96maxplus-u-boot.bin.sd.bin :hk1box-bootloader.img

I previously tested to install to EMMC with selecting 524 from menu without mainline_uboot, just with command "armbian-install" without arguments

**Comment by ophub** [comment]
https://github.com/ophub/amlogic-s9xxx-armbian/commit/797d775ce5dd5472f592575f9326cf25cce4b558

Add mainline_uboot for x96max series

**Comment by Travis90x** [comment]
> [797d775](https://github.com/ophub/amlogic-s9xxx-armbian/commit/797d775ce5dd5472f592575f9326cf25cce4b558)
> 
> Add mainline_uboot for x96max series

My 2101W still won't boot from eMMC after installation with kernel 6.1.15 using DTS for 501 or 502 variant.
With this kernel I boot correctly with USB reader SDcard (DTS 501, 502, 524 ecc...).
Kernel 5.15.99 it's ok, boot from eMMC.

**Comment by Drafteed** [comment]
Hello! 

Having X96 Max + 2101 (4GB / 32GB) my attempts to start the bluetooth were unsuccessful.

Armbian installed in eMMC with mainline_uboot and meson-sm1-x96-max-plus-2101 dtb.

<img width="500" src="https://github.com/ophub/amlogic-s9xxx-armbian/assets/11841379/ee4a4c39-a01b-457d-bf2c-dd0539353a80">

&nbsp;

Board one to one, like @drafort have, but look like 6222B-SRC has a slightly different markings:

<img width="500" src="https://github.com/ophub/amlogic-s9xxx-armbian/assets/11841379/00ad6b11-942b-4bc0-ab42-5af54f3b311d">

&nbsp;

@salsabilng Can you give me an idea how to resolve this?
